bidiServer/server.js

```diff
@@ -208,85 +210,129 @@ async function processCommand(commandData, session) {
-      return await processDebugPageClose(commandData.params, session, response);
-    case "DEBUG.Page.navigate":
-      return await processDebugPageNavigate(commandData.params, session, response);
+      return await process_DEBUG_Page_close(commandData.params, session, response);
     case "DEBUG.Page.screenshot":
-      return await processDebugPageScreenshot(commandData.params, session, response);
+      return await process_DEBUG_Page_screenshot(commandData.params, session, response);
+    case "DEBUG.Page.runJS":
+      return await process_DEBUG_Page_runJS(commandData.params, session, response);
     default:
       throw new Error('unknown command');
   }
 }
 
-async function processDebugBrowserNewPage(params, session, response) {
-  const page = await session.browser.newPage();
+async function process_PROTO_browsingContext_createContext(params, session, response) {
+  const page = await session.browser.newPage(params.url);
 
   // Use CDP targetID for mapping.
   const pageID = page.target()._targetId;
   session.pages[pageID] = page;
 
-  response.result = { pageID };
+  response.result = getBrowsingContextInfo(page.target());
 
   addPageEventHandlers(pageID, page, session.connection);
 
   return response;
 }
 
-async function processDebugPageClose(params, session, response) {
+async function process_PROTO_browsingContext_navigate(params, session, response) {
   const page = getPage(params, session);
-  const pageID = page.target()._targetId;
 
-  page.close();
-  // Remove page from session map.
-  session.pages[pageID] = undefined;
+  if (!params.url) {
+    throw new Error('missing params.url');
+  }
+
+  const availableWaitUntilValues = ["load", "domcontentloaded", "networkidle0", "networkidle2"]
+
+  const options = {};
+  if (params.waitUntil) {
+    if (Array.isArray(params.waitUntil)) {
+      options.waitUntil = [];
+      for (const k in params.waitUntil) {
+        const val = params.waitUntil[k];
+        if (availableWaitUntilValues.includes(val))
+          options.waitUntil.push(val);
+        else
+          throw new Error('unknown params.waitUntil: ' + val);
+      }
+    } else {
+      if (availableWaitUntilValues.includes(params.waitUntil))
+        options.waitUntil = params.waitUntil;
+      else
+        throw new Error('unknown params.waitUntil: ' + params.waitUntil);
+    }
+  }
+
+  if (params.referer) {
+    options.referer = params.referer;
+  }
 
+  if (params.timeout >= 0) {
+    options.timeout = params.timeout;
+  }
+
+  await page.goto(params.url, options);
   response.result = {};
 
   return response;
 }
 
-async function processDebugPageNavigate(params, session, response) {
+async function process_DEBUG_Page_close(params, session, response) {
   const page = getPage(params, session);
+  const pageID = page.target()._targetId;
 
-  if (!params.url) {
-    throw new Error('missing params.url');
-  }
+  page.close();
+  // Remove page from session map.
+  session.pages[pageID] = undefined;
```

Comment:
```suggestion
  delete session.pages[pageID];
```

The difference will be a property with the value undefined, or no property at all if you delete.

But making `session.pages` into a Map would probably make this code clearer, then you'd use `sessions.pages.delete(pageID)`.